static/app/views/dashboardsV2/widgetBuilder/buildSteps/sortByStep/sortBySelectors.tsx

```diff
@@ -54,15 +84,48 @@ export function SortBySelectors({values, sortByOptions, widgetType, onChange}: P
         name="sortBy"
         menuPlacement="auto"
         placeholder={`${t('Select a column')}\u{2026}`}
-        value={values.sortBy}
-        options={uniqBy(sortByOptions, ({value}) => value)}
+        value={showCustomEquation ? CUSTOM_EQUATION_VALUE : values.sortBy}
+        options={[
+          ...uniqBy(sortByOptions, ({value}) => value),
+          ...(hasGroupBy
+            ? [{value: CUSTOM_EQUATION_VALUE, label: t('Custom Equation')}]
+            : []),
+        ]}
         onChange={(option: SelectValue<string>) => {
+          const isSortingByCustomEquation = option.value === CUSTOM_EQUATION_VALUE;
+          setShowCustomEquation(isSortingByCustomEquation);
+          if (isSortingByCustomEquation) {
+            onChange(customEquation);
+            return;
+          }
+
           onChange({
             sortBy: option.value,
             sortDirection: values.sortDirection,
           });
         }}
       />
+      {showCustomEquation && (
+        <ArithmeticInputWrapper>
+          <ArithmeticInput
+            name="arithmetic"
+            key="parameter:text"
```

Comment:
do you need a key here?

Reply:
I don't need a key here. I've removed it!